hail/python/hail/methods/relatedness/king.py

```diff
@@ -0,0 +1,289 @@
+            &= \frac{N^{Aa,Aa}_{i,j} - 2 N^{AA,aa}_{i,j}}
+                    {N^{Aa}_{i} + N^{Aa}_{j}}
+        \end{aligned}
+
+    This "within-family" estimator over-estimates the kinship coefficient under
+    certain circumstances detailed in Section 2.3 of Manichaikul, et. al. The
+    authors recommend an alternative estimator when individuals are known to be
+    from different families. The estimator replaces the average count of
+    heteroyzgous genotypes with the minimum count of heterozygous genotypes:
+
+    .. math::
+
+        \frac{N^{Aa}_{i} + N^{Aa}_{j}}{2} \rightsquigarrow \mathrm{min}(N^{Aa}_{i}, N^{Aa}_{j})
+
+    This transforms the "within-family" estimator into the "between-family"
+    estimator, defined by Equation 11 of Manichaikul, et. al.:
+
+    .. math::
+
+        \begin{aligned}
+        \widetilde{d_{i,j}^{\mathrm{between}}} &=
+            \frac{4 N^{AA,aa}_{i,j} + N^{Aa}_{i} + N^{Aa}_{j} - 2 N^{Aa,Aa}_{i,j}}
+                 {2 \mathrm{min}(N^{Aa}_{i}, N^{Aa}_{j})} \\
+        \widehat{\phi_{i,j}^{\mathrm{between}}} &=
+            \frac{1}{2}
+            + \frac{2 N^{Aa,Aa}_{i,j} - 4 N^{AA,aa}_{i,j} - N^{Aa}_{i} - N^{Aa}_{j}}
+                   {4 \cdot \mathrm{min}(N^{Aa}_{i}, N^{Aa}_{j})}
+        \end{aligned}
+
+    This function, :func:`.king`, only implements the "between-family"
+    estimator, :math:`\widehat{\phi_{i,j}^{\mathrm{between}}}`.
+
+    Parameters
+    ----------
+    call_expr : :class:`.CallExpression`
+        Entry-indexed call expression.
+    block_size : :obj:`int`, optional
+        Block size of block matrices used in the algorithm.
+        Default given by :meth:`.BlockMatrix.default_block_size`.
+
+    Returns
+    -------
+    :class:`.MatrixTable`
+        A :class:`.MatrixTable` whose rows and columns are keys are taken from
+       `call-expr`'s column keys. It has one entry field, `phi`.
+    """
+    mt = matrix_table_source('king/call_expr', call_expr)
+
+    is_hom_ref = Env.get_uid()
+    is_het = Env.get_uid()
+    is_hom_var = Env.get_uid()
+    is_defined = Env.get_uid()
+    mt = mt.select_entries(**{
+        is_hom_ref: hl.float(hl.or_else(call_expr.is_hom_ref(), 0)),
+        is_het: hl.float(hl.or_else(call_expr.is_het(), 0)),
+        is_hom_var: hl.float(hl.or_else(call_expr.is_hom_var(), 0)),
+        is_defined: hl.float(hl.is_defined(call_expr))
+    })
+    ref = hl.linalg.BlockMatrix.from_entry_expr(mt[is_hom_ref], block_size=block_size)
+    het = hl.linalg.BlockMatrix.from_entry_expr(mt[is_het], block_size=block_size)
+    var = hl.linalg.BlockMatrix.from_entry_expr(mt[is_hom_var], block_size=block_size)
+    defined = hl.linalg.BlockMatrix.from_entry_expr(mt[is_defined], block_size=block_size)
+    ref_var = (ref.T @ var).checkpoint(hl.utils.new_temp_file())
+    # We need the count of times the pair is AA,aa and aa,AA. da_ref_var is only AA,aa.
+    # Transposing da_ref_var gives da_var_ref, i.e. aa,AA.
```

Comment:
What does `da_ref_var` refer to? Old name for `ref_var`?

Reply:
yep. fixed